src/target/samd.c

```diff
@@ -773,23 +772,18 @@ static bool samd_set_bootprot(target *t, uint16_t value, const char **argv)
 
 	/* Issue the erase command */
 	target_mem_write32(t, SAMD_NVMC_CTRLA, SAMD_CTRLA_CMD_KEY | SAMD_CTRLA_CMD_ERASEAUXROW);
-
-	/* Poll for NVM Ready */
-	while ((target_mem_read32(t, SAMD_NVMC_INTFLAG) & SAMD_NVMC_READY) == 0) {
-		if (target_check_error(t))
-			return -1;
-	}
+	if (!samd_wait_nvm_ready(t))
+		return false;
 
 	/* Modify the low word of the user row */
-	low = (low & 0xFFFFFFF8) | ((value << 0) & 0x00000007);
+	low = (low & 0xFFFFFFF8U) | ((value << 0U) & 0x00000007U);
```

Comment:
Hex lower case?

Reply:
Good catch! We'll deal with this momentarily